src/test/java/core/basesyntax/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,98 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "kandibober";
+    private static final String VALID_PASSWORD = "ibrahim13";
+    private static final String INVALID_LOGIN = "kandi";
+    private static final String INVALID_PASSWORD = "ibr13";
+    private static final int VALID_AGE = 25;
+    private static final int INVALID_AGE = 10;
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    public void setUp() {
+        Storage.people.clear();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @Test
+    public void register_validUser_ok() {
+        User expected = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        User actual = registrationService.register(expected);
+        assertNotNull(actual);
+        assertEquals(expected.getLogin(), actual.getLogin());
+        assertEquals(expected.getAge(), actual.getAge());
+        assertEquals(expected.getPassword(), actual.getPassword());
+    }
+
+    @Test
+    void register_userIsNull_exceptionThrown() {
+        assertThrows(RegistrationException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    public void register_userWithExistingLogin_exceptionThrown() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        registrationService.register(user);
```

Comment:
Use `Storage.people.add` то imitate existing user in storage instead of `registrationService.register(user)`
<img width="1008" alt="image" src="https://github.com/mate-academy/jv-registration-validation/assets/32339836/7a39d973-7531-4753-b72d-fc1e8fc1421f">